File: src/rust/mkdir.rs
/* Light builtins's (Rust version)
 * mkdir - make directories
 * A lightweight repository of useful shell commands
 * GitHub: https://www.github.com/0x4248/light_builtins
 * Licence: GNU General Public License v3.0
 * By: 0x4248
*/

use std::env;
use std::fs;

fn main() {
    let args: Vec<String> = env::args().collect();
    let path;
    /* Get the command line arguments */
    if args.len() == 1 {
        println!("mkdir: missing operand");
        return;
    } else {
        path = &args[1];
    }
    /* Create the directory */
    match fs::create_dir(path) {
        Result::Ok(_) => (),
        Result::Err(_error) => println!("mkdir: cannot create directory '{}': File exists", path),
    }
}

File: src/python/mkdir.py
#!/usr/bin/env python3
# Light builtins's (Python version)
# mkdir - make directories
# A lightweight repository of useful shell commands
# GitHub: https://www.github.com/0x4248/light_builtins
# Licence: GNU General Public License v3.0
# By: 0x4248

import os
import sys


def main(args):
    for arg in args:
        os.mkdir(arg)
    return 0


if __name__ == "__main__":
    sys.exit(main(sys.argv[1:]))
